cmd/generate-registry-metadata/main.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/sirupsen/logrus"
+)
+
+type options struct {
+	registry string
+}
+
+func (o *options) Validate() error {
+	if o.registry == "" {
+		return errors.New("--registry is required")
+	}
+	return nil
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&o.registry, "registry", "", "Path to the step registry directory.")
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Fatal("could not parse input")
+	}
+	return o
+}
+
+func updateMetadata(registryPath string) (api.RegistryMetadata, error) {
+	metadata := api.RegistryMetadata{}
+	metadata.ComponentPath = make(map[string]string)
+	if err := filepath.Walk(registryPath, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info != nil && !info.IsDir() && filepath.Ext(info.Name()) == ".yaml" {
+			relpath, err := filepath.Rel(registryPath, path)
+			if err != nil {
+				return fmt.Errorf("failed to determine relative path for %s: %w", path, err)
+			}
+			metadata.ComponentPath[info.Name()] = filepath.Dir(relpath)
```

Comment:
If this is all we store, we will need to do GitHub (?) calls to get the OWNERS? Or are you expecting to clone the repo? Why not store a 

```go
type info struct {
    Path string
    Owners owners.FullConfig
}

var metadata map[string]info

Reply:
I was originally planning to just get the `OWNERS` files via the `raw.githubusercontent` pages, but that was mainly because I didn't want to clutter the component files too much since it could be confusing for users. Since this is now being handled as a separate metadata file, it would make sense to store them in the metadata.